src/mixxx.cpp

```diff
@@ -598,6 +596,11 @@ void MixxxMainWindow::initializeWindow() {
     Pal.setColor(QPalette::Background, MenuBarBackground);
     m_pMenuBar->setPalette(Pal);
 
+    // restore window size/position
+    QSettings settings("mixxxdj", "mixxx");
```

Comment:
> What happens when we want to migrate to a new profile format a year down the line?

Why would you need to do anything with the widget states after changing to a new profile format? If they become invalid for some reason then just delete them. Deletion is completely acceptable for geometry state, and most other widget states that I can think of storing.

> It would be nice to include this fix in Mixxx if you have the time to improve it.

I'm happy to make small changes to the PR to accommodate your personal preferences, and have pushed a new commit cleaning up the app closing logic to be more inline with the existing code base as you requested. 

If you still insist on storing the widget state in the config file, you'll need to be more explicit about exactly how you want to encode the byte arrays returned by `saveGeometry()` and `saveState()` in the config file. QSettings already has an API to persist such state information, so I'm having difficulty understanding your opposition to using it.